ddpui/ddpdbt/dbt_service.py

```
@@ -937,3 +1003,194 @@ def cleanup_unused_sources(org: Org, orgdbt: OrgDbt, manifest_json=None):
+    orgdbt = org.dbt
+
+    logger.info(
+        f"Switching git repository for org {org.slug} from {orgdbt.gitrepo_url} to {payload.gitrepoUrl}"
+    )
+
+    # Determine which PAT to use for verification
+    is_token_masked = set(payload.gitrepoAccessToken.strip()) == set("*")
+
+    if is_token_masked:
+        if not orgdbt.gitrepo_access_token_secret:
+            raise Exception("Cannot use masked token - no existing PAT found")
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)